<launch>
    <arg name="frame_id"/>
    <arg name="image_topic"/>
    <arg name="camera_info_topic"/>
    <arg name="with_aruco" default="false"/>
    <arg name="with_mediapipe" default="false"/>
    <arg name="visualize" default="true"/>
    <arg name="aruco_with_tracking" default="false"/>

    <node name="tracking_$(arg frame_id)" pkg="camera_tracking" type="tracking_node.py" output="screen" respawn="false">
        <param name="frame_id" type="str" value="$(arg frame_id)"/>
        <param name="image_topic" type="str" value="$(arg image_topic)"/>
        <param name="camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
        <param name="with_aruco" type="bool" value="$(arg with_aruco)"/>
        <param name="with_mediapipe" type="bool" value="$(arg with_mediapipe)"/>
        <param name="visualize" type="bool" value="$(arg visualize)"/>
        <param name="aruco_with_tracking" type="bool" value="$(arg aruco_with_tracking)"/>
    </node>

</launch>
